command/base.go

```diff
@@ -582,8 +584,17 @@ func (f *FlagSets) Completions() complete.Flags {
 }
 
 // Parse parses the given flags, returning any errors.
+// Warnings, if any, regarding the arguments format are sent to stdout
 func (f *FlagSets) Parse(args []string) error {
-	return f.mainSet.Parse(args)
+	err := f.mainSet.Parse(args)
+
+	var out bytes.Buffer
+	printArgsWarningIfAny(&out, f.Args())
+	if out.String() != "" {
+		f.ui.Warn(out.String())
+	}
```

Comment:
Alternatively, you can have `printArgsWarningIfAny` return the warning as a string, and output the string if it's non-empty.